### PR TITLE
Creation of `load.py` functionality

### PR DESCRIPTION
**Pull request by cshinaver**
@mperez14 This is the branch for working on the load stuff

### CODE REVIEW

File: load.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import csv
+import sys
+
+from bookfinder.models.book import Book
+from bookfinder.models.purchasechoice import PurchaseChoice
+from bookfinder.api.scraper import get_books_for_book_title_using_google_books
+from bookfinder.api.scraper import AmazonScraper
+#from bookfinder.api.scraper import get_amazon_purchase_choices_for_keyword
```
Comment (cshinaver) [inline load.py]: remove

File: load.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            ISBN_load = row[1]
+            class_name = row[2]
+
+            # add book object
+            b = Book()  # create temp book object
+            temp_arr = get_books_for_book_title_using_google_books(
+                    'isbn:' + ISBN_load
+                    )
+            if len(temp_arr) == 0:
+                #print "skipping ISBN: "+ISBN_load
```
Comment (cshinaver) [inline load.py]: remove commented stuff

File: load.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                    print "skipping ISBN: "+ISBN_load
+                    continue
+                temp_book = Book()
+                temp_book.isbn = amazon_dict[0].get('isbn')
+                temp_book.title = amazon_dict[0].get('title')
+                temp_book.author = amazon_dict[0].get('author')
+                temp_book.thumbnail_link = amazon_dict[0].get('thumbnail_link')
+                temp_book.subtitle = ''
+                temp_arr.append(temp_book)
+                #continue
```
Comment (cshinaver) [inline load.py]: remove

File: load.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            b.thumbnail_link = temp_var.thumbnail_link
+            query_book = Book.get(
+                    isbn=b.isbn
+                    )  # check if Book object is already in DB
+            if not query_book:
+                b.save()
+            else:
+                b.id = query_book.id
+
+            #print "author "+b.author
```
Comment (cshinaver) [inline load.py]: remove

**Comment by mperez14** [comment]
Having trouble saving BooksViewed. Trouble occurs when saving user_id. Any idea on whats causing the issue?

**Comment by cshinaver** [comment]
The error you're getting is at line 79

**Comment by cshinaver** [comment]
I got this error
```
$ python load.py ISBN_data.csv
['-1', '9780538497879', 'calc 3']
['1', '9781118230640', 'physics 2']
['1', '9780133126013', 'intro to computing programming']
['1', '9780133760033', 'intro to EE 20224']
['1', '9780321982384', 'linear algebra']
['1', '9781118061145', 'physics 3']
['1', '9780195116632', 'electronics']
['1', '9780074637364', 'electronics']
['1', '9780072369779', 'semiconductors']
['1', '0201128195', 'electromagnetic fields']
[]
['-2', '0470458313', 'diff EQ']
['2', '0133943283', 'signals and systems']
['2', '9780072369779', 'semiconductors']
['2', '9781577667834', 'electromagnetic fields']
['2', '9780132151498', 'Photonics']
['2', '9781111425777', 'electric machinery']
['2', '9780470631553', 'microwave circuit design']
['2', '9780123850591', 'computer networks']
['2', '9780199339136', 'Microelectronic circuit design']
['2', '9780132129480', 'data structures']
[]
['-3', '0470458313', 'diff EQ']
['3', '0133943283', 'signals and systems']
Traceback (most recent call last):
  File "load.py", line 90, in <module>
    load_csv(filename)
  File "load.py", line 60, in load_csv
    b.save()    #save book
  File "/Users/charlesshinaver/Dropbox/fallSemester2015/databases/bookfinder/bookfinder/models/base.py", line 169, in save
    _save_as_new_object()
  File "/Users/charlesshinaver/Dropbox/fallSemester2015/databases/bookfinder/bookfinder/models/base.py", line 129, in _save_as_new_object
    id = execute_sql_query(query)[0][0]
  File "/Users/charlesshinaver/Dropbox/fallSemester2015/databases/bookfinder/bookfinder/db/connect.py", line 105, in execute_sql_query
    cursor.execute(query)
  File "/Users/charlesshinaver/.virtualenvs/bookfinder/lib/python2.7/site-packages/psycopg2/extras.py", line 120, in execute
    return super(DictCursor, self).execute(query, vars)
psycopg2.IntegrityError: duplicate key value violates unique constraint "book_isbn_key"
DETAIL:  Key (isbn)=(9780133943283) already exists.
```

**Comment by mperez14** [comment]
am getting this error

matthews-mbp:BookFinder Matthew$ python load.py ISBN_data.csv 
['-1', '9780538497879', 'calc 3']
Traceback (most recent call last):
  File "load.py", line 92, in <module>
    load_csv(filename)
  File "load.py", line 83, in load_csv
    booksV.save()
  File "/Users/Matthew/BookFinder/bookfinder/models/base.py", line 169, in save
    _save_as_new_object()
  File "/Users/Matthew/BookFinder/bookfinder/models/base.py", line 129, in _save_as_new_object
    id = execute_sql_query(query)[0][0]
  File "/Users/Matthew/BookFinder/bookfinder/db/connect.py", line 105, in execute_sql_query
    cursor.execute(query)
  File "/usr/local/lib/python2.7/site-packages/psycopg2/extras.py", line 120, in execute
    return super(DictCursor, self).execute(query, vars)
psycopg2.IntegrityError: insert or update on table "booksviewed" violates foreign key constraint "booksviewed_user_id_fkey"
DETAIL:  Key (user_id)=(1) is not present in table "bookfinderuser".

**Comment by mperez14** [comment]
Scraper is very inconsistent. Just testing the basic search_for_keyword function on amazon sometimes returns a result and other times doesn't. I think for just loading data into DB we should just run the load.py over the ISBN_data several times to ensure that all the books are added. I spent last night trying to fix but I am not getting anywhere

screen shot 2015-12-08 at 2 49 58 pm

**Comment by cshinaver** [comment]
approved

**Pull request merged**